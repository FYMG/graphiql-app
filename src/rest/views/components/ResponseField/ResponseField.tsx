import React from 'react';
import Editor from '@monaco-editor/react';

interface ResponseFieldProps {
  response: Record<string, unknown> | null;
  status: number | null;
}

function ResponseField({ status, response }: ResponseFieldProps) {
  const textColorClass = status === 200 ? 'text-green-500' : 'text-orange-500';

  return (
    <div className="mb-4">
      <h3 className="mb-2 font-semibold">Response</h3>
      {status && (
        <>
          <p className={textColorClass}>Status: {status}</p>
          <Editor
            className="border px-1"
            language="json"
            theme="vs-light"
            value={JSON.stringify(response, null, 2)}
            options={{ readOnly: true, automaticLayout: true }}
            height="100px"
            width="100%"
            data-testid="response-editor"
          />
        </>
      )}
    </div>
  );
}

export default ResponseField;
