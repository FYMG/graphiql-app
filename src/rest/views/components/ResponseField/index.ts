export { default as BodyEditor } from './ResponseField';
