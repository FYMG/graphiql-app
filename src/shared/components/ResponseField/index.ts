export { default as ResponseField } from './ResponseField';
