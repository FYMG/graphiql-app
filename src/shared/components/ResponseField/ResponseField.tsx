import React from 'react';
import { Editor } from '@monaco-editor/react';
import { useTranslations } from 'next-intl';

import { EmptyResponseField } from '@rest/views/components/EmptyResponseField';

interface ResponseFieldProps {
  loading: boolean;
  response: Record<string, unknown> | null;
  status: number | null;
}

function ResponseField({ status, response, loading }: ResponseFieldProps) {
  const t = useTranslations('rest');
  const textColorClass =
    status && status >= 200 && status < 300 ? 'text-green-500' : 'text-orange-500';

  const getEditorValue = () => {
    if (loading) {
      return `${t('loading-response')}...`;
    }

    if (typeof response === 'string') {
      return response;
    }

    return JSON.stringify(response, null, 2);
  };

  return (
    <div className="mb-4 px-1">
      <h3 className="mb-2 font-semibold">{t('response')}</h3>
      {status ? (
        <>
          <p>
            {t('status')}

            <span className={textColorClass}>{loading ? t('loading') : status}</span>
          </p>
          <Editor
            className="border px-1"
            language="json"
            theme="vs-light"
            value={getEditorValue()}
            options={{ readOnly: true, automaticLayout: true }}
            height="150px"
            width="100%"
            data-testid="response-editor"
          />
        </>
      ) : (
        <EmptyResponseField />
      )}
    </div>
  );
}

export default ResponseField;
