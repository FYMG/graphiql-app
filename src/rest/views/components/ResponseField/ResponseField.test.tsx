import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ResponseField from './ResponseField';

const testId = 'response-editor';

describe('ResponseField', () => {
  it('renders with null response and status', () => {
    render(<ResponseField response={null} status={null} loading={false} />);

    expect(screen.getByRole('heading', { name: /Response/i })).toBeInTheDocument();
    expect(screen.queryByText(/Status/i)).not.toBeInTheDocument();
    expect(screen.queryByTestId(testId)).not.toBeInTheDocument();
  });

  it('matches snapshot when response is null, loading is false, and status is 200', () => {
    const { asFragment } = render(
      <ResponseField loading={false} response={null} status={200} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
