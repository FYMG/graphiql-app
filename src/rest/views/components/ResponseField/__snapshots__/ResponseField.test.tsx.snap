// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseField matches snapshot when response is null, loading is false, and status is 200 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <h3
      class="mb-2 font-semibold"
    >
      Response
    </h3>
    <p>
      Status:
      <span
        class="text-green-500"
      >
        200
      </span>
    </p>
    <section
      style="display: flex; position: relative; text-align: initial; width: 100%; height: 150px;"
    >
      <div
        style="display: flex; height: 100%; width: 100%; justify-content: center; align-items: center;"
      >
        Loading...
      </div>
      <div
        class="border px-1"
        style="width: 100%; display: none;"
      />
    </section>
  </div>
</DocumentFragment>
`;
